msticpy/vis/entity_graph_tools.py

```diff
@@ -0,0 +1,559 @@
+            {
+                "Name": names,
+                "Description": descs,
+                "Type": types,
+                "TimeGenerated": times,
+                "EndTime": endtimes,
+                "StartTime": starttimes,
+            }
+        )
+        tl_df.replace("None", np.NaN, inplace=True)
+        return tl_df
+
+    def _add_incident_or_alert_node(self, incident: Union[Incident, Alert, None]):
+        """Check what type of entity is passed in and creates relevent graph."""
+        if isinstance(incident, Incident):
+            self._add_incident_node(incident)
+        elif isinstance(incident, Alert):
+            self._add_alert_node(incident)
+
+    def _add_entity_node(self, ent, attached_to=None):
+        """Add an Entity to the graph."""
+        self.alertentity_graph = nx.compose(self.alertentity_graph, ent.to_networkx())
+        if attached_to:
+            self.add_link(attached_to, ent.name_str)
+
+    def _add_alert_node(self, alert, incident_name=None):
+        """Add an alert entity to the graph."""
+        self.alertentity_graph = nx.compose(self.alertentity_graph, alert.to_networkx())
+        if alert["Entities"]:
+            for ent in alert["Entities"]:
+                self._add_entity_node(ent, alert.name_str)
+        if incident_name:
+            self.add_link(incident_name, alert.name_str)
+
+    def _add_incident_node(self, incident):
+        """Add an incident entity to the graph."""
+        self.alertentity_graph = nx.compose(
+            self.alertentity_graph, incident.to_networkx()
+        )
+        if incident.Alerts:
+            for alert in incident.Alerts:
+                self._add_alert_node(alert, incident.name_str)
+        if incident.Entities:
+            entities = _dedupe_entities(incident.Alerts, incident.Entities)
+            for ent in entities:
+                self._add_entity_node(ent, incident.name_str)
+
+    def _add_entity_edges(self, edges: set, attached_to: str):
+        """Check entity edges and add them."""
+        for edge in edges:
+            if isinstance(edge.target, Entity):
+                if not self.alertentity_graph.has_node(edge.target.name_str):
+                    self._add_entity_node(edge.target)
+                try:
+                    self.add_link(attached_to, edge.target.name_str)
+                except MsticpyUserError:
+                    pass
+
+    @property
+    def graph(self) -> nx.Graph:
+        """Return the raw NetworkX graph."""
+        return self.alertentity_graph
+
+
+def _dedupe_entities(alerts, ents) -> list:
+    """Deduplicate incident and alert entities."""
+    alrt_ents = []
+    for alrt in alerts:
+        if alrt["Entities"]:
+
+            alrt_ents += [ent.__hash__() for ent in alrt["Entities"]]
+    for ent in ents:
+        if ent.__hash__() in alrt_ents:
+            ents.remove(ent)
+    return ents
+
+
+@pd.api.extensions.register_dataframe_accessor("mp_incident_graph")
```

Comment:
I think I mentioned this before somewhere. 
I was hoping to start consolidating our pandas extensions so that we don't have one for every features. The plotting ones are easiest - since they can be nicely grouped under an mp_plot. I think I did this for the matrix plot - it doesn't have its own accessor.
We should deprecate the other (timeline, etc) and remove them at some point.
The only reason to keep a separate accessor like this is if you need something very specific to the entity_graph, ,ike the build_graph function.
Are you OK with removing this and just relying on mp_plot?